src/com/dotcms/publisher/environment/business/EnvironmentAPIImpl.java

```diff
@@ -63,12 +64,17 @@ public void deleteEnvironment(String id) throws DotDataException {
 			return;
 
 		// remove the endpoints of the environment
-
+		
 		List<PublishingEndPoint> endPoints = APILocator.getPublisherEndPointAPI().findSendingEndPointsByEnvironment(id);
 
-		for (PublishingEndPoint ep : endPoints) {
-			APILocator.getPublisherEndPointAPI().deleteEndPointById(ep.getId());
-		}
+        IntegrityUtil integrityUtil = new IntegrityUtil();
+        
+        for (PublishingEndPoint ep : endPoints) {
+            //Delete Existing conflicts reported for this endpoint
+            integrityUtil.completeDiscardConflicts(ep.getId());
```

Comment:
I suggest moving this call inside APILocator.getPublisherEndPointAPI().deleteEndPointById. That way it won't need to be repeated in the REST Resource and also guarantees that a direct call to APILocator.getPublisherEndPointAPI().deleteEndPointById also takes care of this.